documentation/docker.md

```diff
@@ -0,0 +1,229 @@
+    `--interactive` ("Keep STDIN open even if not attached")
+    `--tty` ("Allocate a pseudo-TTY")
+- `--rm` - tells the Docker engine to remove the container (but not the image) when it exits
+- `/bin/sh` - starts a shell (as the image is based off Alpine Linux, you'll get an error if you try to start `/bin/bash`)
+
+At this point you'll be in the `/teacher-vacancy` directory
+
+## Advanced features
+
+### BuildKit builds
+
+- [BuildKit](https://docs.docker.com/develop/develop-images/build_enhancements/) was introduced in Docker 18.09
+- Enabling BuildKit sped up builds by at least a minute
+
+### Buildx
+
+- [Buildx](https://docs.docker.com/buildx/working-with-buildx/) was introduced in Docker 19.03
+- This is a requirement for the newer GitHub Action
+
+### Caching
+
+- Pulling an image, or image layer from the Docker Hub repository, is typically much less "expensive" in time than building from source
+- We cache images for the `builder` and `production` targets, across all branches.
+
+### `Builder` stage image tags
+
+- `builder-dev`
+- `builder-staging`
+- `builder-master`
+
+And for feature branches, these may undergo several pushes to the branch, so it's worth storing the builder image, to speed up subsequent builds, e.g. 
+
+- `builder-TEVA-1296-alert-filters`
+
+### `Production` stage image branch tags
+
+- `dev`
+- `staging`
+- `master`
+
+And for feature branches: 
+
+- `TEVA-1296-alert-filters`
+
+The image tied to these tags changes frequently, with each build, for caching. 
+
+Do NOT use these tags to generate containers - instead, choose a unique tag listed below
+
+### `Production` stage image unique tags
+
+For images built off the `master` branch, we use the [SHA of the GitHub commit](https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions), e.g.
+
+[Docker image tagged `a18165a5a6d8ae5b753ac7c3cac65f0cbc34dd18`](https://hub.docker.com/layers/dfedigital/teaching-vacancies/a18165a5a6d8ae5b753ac7c3cac65f0cbc34dd18/images/sha256-7ae9ec3802192d41dc9846db734303efecc372d90d11317864f2cc0478908bf5?context=explore) comes from [GitHub commit `a18165a5a6d8ae5b753ac7c3cac65f0cbc34dd18`](https://github.com/DFE-Digital/teacher-vacancy-service/commit/a18165a5a6d8ae5b753ac7c3cac65f0cbc34dd18)
+
+
+### Multi-stage builds
+
+- [Multi-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/) were introduced in Docker 17.05
+
+## Docker Hub
+
+### Docker Hub organisation
+
+The Docker images for Teaching Vacancies are stored in the Docker Hub organisation [`dfedigital`](https://hub.docker.com/u/dfedigital).
+
+### Docker Hub team
+
+The Docker Hub team [`teacherservices`](https://hub.docker.com/orgs/dfedigital/teams/teacherservices) has multiple members, (~35 in October 2020), each of which is a Docker Hub user.
+
+### Docker Hub users
+
+The user `teachingjobs` is used as a service account, by GitHub Actions to publish images.
+Credentials for `teachingjobs`:
```

Comment:
It may be worth addind which email it is mapped to and who has access to the email in case the dockerhub password must be reset

Reply:
Added details of the email address